src/api/events/index.js

```diff
@@ -1,16 +1,38 @@
+const AWSXRay = require('aws-xray-sdk');
 const handleWorkRequest = require('../event-services/work-request');
 const logger = require('../../utils/logging');
 
-
 module.exports = (socket) => {
-  socket.on('WorkRequest', async (data) => {
-    logger.log('Work submitted from client', socket.id, ':', data);
+  socket.on('WorkRequest', (data) => {
+    const segment = new AWSXRay.Segment('API-development-socket.io');
+    const ns = AWSXRay.getNamespace();
+
+    ns.runPromise(async () => {
+      AWSXRay.capturePromise();
+      AWSXRay.setSegment(segment);
+
+      logger.log('Work submitted from client', socket.id, ':', data);
+
+      segment.addIncomingRequestData({
+        request: {
+          method: 'POST',
+          url: 'socketio://WorkRequest',
+        },
+      });
+
+      const { uuid, experimentId } = data;
+      segment.addAnnotation('uuid', uuid);
+      segment.addAnnotation('experimentId', experimentId);
+
+      try {
+        await handleWorkRequest(data, socket);
+      } catch (e) {
+        logger.error('Error while processing WorkRequest event:');
+        logger.trace(e);
+        segment.addError(e);
+      }
 
-    try {
-      await handleWorkRequest(data, socket);
-    } catch (e) {
-      logger.error('Error while processing WorkRequest event:', e);
-      logger.trace(e);
-    }
+      segment.close();
+    });
```

Comment:
The whole handler for the SocketIO Work Request had to be wrapped in this thing for XRay to recognize it. Other than that, there is no modification in the code.